### PR TITLE
[WIP] Improvements to tomography and qi tools modules

### PR DESCRIPTION
**Pull request by chriseclectic**
## Description
Improves tomography tools module to allow:

- Default SIC-POVM based initial state for process tomography.
- Custom input and measurement bases
- Partial tomography
- Unifying state and process tomography functions by using a dict of experimental configurations

Added some features to qi module:

- partial trace more efficient if input is a state vector
- Added several entropy and mutual information functions
- Added functions for generating random density matrices

Fixes some bugs in the `qi` module.

## Motivation and Context
Current implementation of process tomography is suboptimal in the number of prepared initial states.

## How Has This Been Tested?
Basic testing in notebooks has done, but proper unit tests still need to be written.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

Wrapper functions were added to maintain backwards compatibilty with old tomography.py though these should either be removed or phased out.

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: qiskit/tools/qi/qi.py [hunk 1]
```diff
@@ -288,6 +295,99 @@ def outer(v1, v2=None):
...
+        method (string) optional: the method to use.
+
+    Returns:
+        rho (N,N) ndarray: a density matrix.
+
+    method:
+        'hs': sample rho from the Hilbert-Schmidt metric.
+        'bures': sample rho from the Bures metric.
+    """
+    if method == 'hs':
```
Comment (atilag) [inline qiskit/tools/qi/qi.py]: 'hs' => 'hilbert-schmidt'

File: qiskit/tools/qi/qi.py [hunk 2]
```diff
@@ -288,6 +295,99 @@ def outer(v1, v2=None):
...
+        return __random_density_hs(N, rank)
+    elif method == 'bures':
+        return __random_density_bures(N, rank)
+    else:
+        print('Error: unrecognized method ', method)
```
Comment (atilag) [inline qiskit/tools/qi/qi.py]: `print()` statements are not allowed. We better throw a `QISKitError`:
```python
else:
    raise QISKitError('Error: unrecognized method {}'.format(method))
```

**Comment by chriseclectic** [comment]
@atilag I made the changes to the `qi.py` file

**Comment by chriseclectic** [comment]
I'm closing this WIP pull request and resubmitting as two separate pull requests: one for the qi module, and one for tomography module.

**Pull request closed without merging**